packages/pigeon/platform_tests/shared_test_plugin_code/lib/integration_tests.dart

```diff
@@ -39,6 +39,107 @@ enum TargetGenerator {
 void runPigeonIntegrationTests(TargetGenerator targetGenerator) {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
 
+  bool compareAllTypes(AllTypes? allTypesOne, AllTypes? allTypesTwo) {
```

Comment:
Could you document these? The combination of `expect` and a return boolean is confusing to me. Why do some differences return `false`, while others fail the test directly? Could it just return `void` and have the one `false` case be an expect failure instead?

Reply:
I didn't love it either. I'm going to improve it...

Reply:
I have to check for null before comparing the classes, to avoid crashing or potential bad tests (ie. nullClass? == class.null? would pass when it shouldn't). I can't use expect() in that instance. 

I could change everything to if -> return. That was just a lot messier. Probably a better approach though.

Reply:
Why can't you use `expect` for the null check?

```
if (allTypesOne == null && allTypesTwo == null) {
  return;
}

expect(allTypesOne != null && allTypesTwo != null, true);
```

Or even just:
```
expect(allTypesOne == null, allTypesTwo == null);
```

Reply:
> expect(allTypesOne == null, allTypesTwo == null);

Seems so obvious now...